src/compressed-size.js

```
@@ -8,6 +9,12 @@ const getCompressedSize = (data, compression = 'gzip') => {
       size = gzip.sync(data)
       break
     case 'brotli':
+      try {
+        brotli = require('brotli-size')
+      } catch (e) {
+        throw new Error(`Missing optional dependency. Install it with:
```

I don't think you should throw an error here. I would say, the travis build should pass because it only invokes bundlesize and the bundlesize build status should error out with the right message.

This should help: https://github.com/siddharthkp/bundlesize/blob/master/src/build.js#L32

---

I'm not sure I understand. Do you want to make the change on this branch?

---

yep, this branch

---

I think siddharthkp was saying that you can make bundlesize exit with non-zero (so travis doesn't fail, but report bundlesize failure status to GitHub

In an offshoot version of bundlesize, I've gone with failing the build 
https://github.com/bundlewatch/bundlewatch/blob/master/src/app/getLocalFileDetails/getSize.js#L12

---

I'd suggest the way to fail the bundlesize build would be https://github.com/siddharthkp/bundlesize/blob/master/src/build.js#L32

---

@siddharthkp Would you like to implement this?

---

I don't mind, it might take me some time to get to this though 😅